app/lib/submission_builder/document.rb

```diff
@@ -114,6 +114,34 @@ def email_from_intake_or_df
       end
     end
 
+    def process_md_mailing_street(xml)
+      return unless @submission.data_source.direct_file_data.mailing_street.present?
+
+      mailing_street = sanitize_for_xml(@submission.data_source.direct_file_data.mailing_street)
+      key_position = mailing_street.index(ADDRESS_ABBREV_REGEX)
+
+      if key_position
+        truncated_mailing_street = mailing_street[0, key_position].rstrip
+        excess_characters = mailing_street[key_position..].lstrip
+
+        xml.AddressLine1Txt sanitize_for_xml(truncated_mailing_street, 35)
+        if @submission.data_source.direct_file_data.mailing_apartment.present?
+          apartment = sanitize_for_xml(@submission.data_source.direct_file_data.mailing_apartment)
+          if apartment.length + excess_characters.length > 35
+            truncated_apartment = apartment[0, 35 - excess_characters.length].rpartition(' ').first
+            xml.AddressLine2Txt "#{excess_characters} #{truncated_apartment}"
+          else
+            xml.AddressLine2Txt "#{excess_characters} #{apartment}"
+          end
+        else
+          xml.AddressLine2Txt excess_characters
+        end
+      else
+        xml.AddressLine1Txt sanitize_for_xml(mailing_street, 35)
+        xml.AddressLine2Txt sanitize_for_xml(@submission.data_source.direct_file_data.mailing_apartment, 35) if @submission.data_source.direct_file_data.mailing_apartment.present?
+      end
+    end
+
     def process_mailing_street(xml)
```

Comment:
a comment here noting that these are NY-only would help clarify the difference from the new code above